pennylane/pulse/transmon.py

```diff
@@ -215,3 +217,263 @@ def __add__(self, other):
+    Let us look at a chain of three transmon qubits that are coupled with their direct neighbors. We provide all numbers in
+    units of :math:`2\pi\text{GHz}`. We parametrize the amplitude as a sinusodial and make the maximum amplitude
+    as well as the drive frequency trainable parameters. We simulate the evolution for a time window of :math:`[0, 5]\text{ns}`.
+
+    .. code-block:: python3
+
+        omega = [5.1, 5., 5.3]
+        connections = [[0, 1], [1, 2]]
+        g = [0.02, 0.05]
+        H = qml.pulse.transmon_interaction(omega, connections, g, wires=range(3))
+
+        def amp(Omega, t): return Omega * jnp.sin(t)
+        def freq(fr, t): return fr
+        phase = 0.
+        t=2
+
+        H += qml.pulse.transmon_drive(amp, phase, freq, 0)
+        H += qml.pulse.transmon_drive(amp, phase, freq, 1)
+        H += qml.pulse.transmon_drive(amp, phase, freq, 2)
+
+        dev = qml.device("default.qubit.jax", wires=range(3))
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def qnode(params):
+            qml.evolve(H)(params, t=5.)
+            return qml.expval(qml.PauliZ(0) + qml.PauliZ(1) + qml.PauliZ(2))
+
+    We evaluate the Hamiltonian with some arbitrarily chosen maximum amplitudes and set
+    the drive frequency equal to the qubit frequencies. Note how the order of the construction
+    of ``H`` determines the order with which the parameters need to be passed to
+    :class:`~.ParametrizedHamiltonian` and :func:`~.evolve`. We made the drive frequencies
+    trainable parameters by providing a constant callable above instead of fixed values.
+    This allows us to differentiate with respect to the frequencies and to optimize them.
+
+    >>> Omega0, Omega1, Omega2 = [0.5, 0.3, 0.6]
+    >>> fr0, fr1, fr2 = omega
+    >>> params = [Omega0, fr0, Omega1, fr1, Omega2, fr2]
+    >>> qnode(params)
+    Array(2.25098131, dtype=float64)
+    >>> jax.grad(qnode)(params)
+    [Array(-0.96356123, dtype=float64),
+     Array(-0.0189564, dtype=float64),
+     Array(-0.58581467, dtype=float64),
+     Array(0.24023855, dtype=float64),
+     Array(-1.30009675, dtype=float64),
+     Array(-0.10709503, dtype=float64)]
+
+    """
+    if d != 2:
+        raise NotImplementedError(
+            "Currently only supports qubits (d=2). Qutrits and qudits support is planned in the future."
+        )
+
+    wires = Wires(wires)
+
+    # TODO: use creation and annihilation operators when introducing qutrits
+    # We compute the `coeffs` and `observables` of the EM field
+    coeffs = [
+        AmplitudeAndPhaseAndFreq(qml.math.cos, amplitude, phase, freq),
+        AmplitudeAndPhaseAndFreq(qml.math.sin, amplitude, phase, freq),
+    ]
+
+    drive_x_term = sum(qml.PauliX(wire) for wire in wires)
+    drive_y_term = sum(-qml.PauliY(wire) for wire in wires)
+
+    observables = [drive_x_term, drive_y_term]
+
+    return HardwareHamiltonian(coeffs, observables, reorder_fn=_reorder_AmpPhaseFreq)
```

Comment:
missing the `HardwarePulse` storing the drive parameters

Reply:
Oh good catch!

Reply:
fixed ✅